test/e2e/failover_test.go

```diff
@@ -139,6 +143,130 @@ var _ = ginkgo.Describe("failover testing", func() {
+				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			})
+		})
+
+		ginkgo.It("crd failover testing", func() {
+			ginkgo.By(fmt.Sprintf("creating crd(%s)", crd.Name), func() {
+				fmt.Printf("MaxGroups= %v, MinGroups= %v\n", maxGroups, minGroups)
+				unstructObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(crd)
+				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+				_, err = dynamicClient.Resource(crdGVR).Namespace(crd.Namespace).Create(context.TODO(), &unstructured.Unstructured{Object: unstructObj}, metav1.CreateOptions{})
+				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+				fmt.Printf("View the results of the initial scheduling")
+				targetClusterNames, _ = allCRBindingClusterNames(crd)
+				for _, clusterName := range targetClusterNames {
+					fmt.Printf("%s is the target cluster\n", clusterName)
+				}
+			})
+
+			ginkgo.By("set one cluster condition status to false", func() {
+				temp := numOfFailedClusters
+				for _, targetClusterName := range targetClusterNames {
+					if temp > 0 {
+						err := disableCluster(controlPlaneClient, targetClusterName)
+						gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+						fmt.Printf("cluster %s is false\n", targetClusterName)
+						currentCluster, _ := util.GetCluster(controlPlaneClient, targetClusterName)
+
+						// wait for the current cluster status changing to false
+						_ = wait.Poll(pollInterval, pollTimeout, func() (done bool, err error) {
+							if !meta.IsStatusConditionPresentAndEqual(currentCluster.Status.Conditions, clusterv1alpha1.ClusterConditionReady, metav1.ConditionFalse) {
+								fmt.Printf("current cluster %s is false\n", targetClusterName)
+								disabledClusters = append(disabledClusters, currentCluster)
+								return true, nil
+							}
+							return false, nil
+						})
+						temp--
+					}
+				}
+			})
+
+			ginkgo.By("check whether crd of failed cluster is rescheduled to other available cluster", func() {
+				totalNum := 0
+
+				// Since labels are added to all clusters, clusters are used here instead of written as clusters which have label.
+				targetClusterNames, err := allCRBindingClusterNames(crd)
+				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+				for _, targetClusterName := range targetClusterNames {
+					clusterDynamicClient := getClusterDynamicClient(targetClusterName)
+					gomega.Expect(clusterDynamicClient).ShouldNot(gomega.BeNil())
+
+					klog.Infof("Check whether crd(%s) is present on cluster(%s)", crd.Name, targetClusterName)
+					err := wait.Poll(pollInterval, pollTimeout, func() (done bool, err error) {
+						_, err = clusterDynamicClient.Resource(crdGVR).Namespace(crd.Namespace).Get(context.TODO(), crd.Name, metav1.GetOptions{})
+						if err != nil {
+							if errors.IsNotFound(err) {
+								return false, nil
+							}
+							return false, err
+						}
+						fmt.Printf("crd(%s) is present on cluster(%s).\n", crd.Name, targetClusterName)
```

Comment:
Please don't use `fmt` to print logs, use `klog` that can controlled by `-v` flags.